Automattic/tap-zendesk
# Including minimum remain rate limit config

# Description of change
When the Zendesk API request rate limit is shared with different applications, using all the limits available could impact the correct behavior of these applications. So this PR includes a configurable parameter that sets a soft limit for the remains request rate limit and avoids consuming all request rate limits available.

# Manual QA steps
 - Executed the extractor with a high `min_remain_rate_limit` of almost the rate limit available (e.g. 630 if the rate limit is 700)
 - Ran the extractor and checked the log to verify if the reset time is being respected.
```
2022-12-20T21:42:38.729545Z [info     ] time=2022-12-20 18:42:38 name=singer level=INFO message=METRIC: {"type": "timer", "metric": "http_request_duration", "value": 0.31553101539611816, "tags": {"endpoint": "https://woothemes.zendesk.com/api/v2/tickets/5805588/metrics.json", "status": "succeeded"}} cmd_type=extractor name=tap-zendesk run_id=97154896-0092-4a8e-8390-63d337f5bac6 state_id=2022-12-20T214210--tap-zendesk--target-jsonl stdio=stderr
2022-12-20T21:42:38.729954Z [info     ] time=2022-12-20 18:42:38 name=singer level=INFO message=API rate limit exceeded (rate limit: 700, remain: 630, min remain limit: 630). Tap will retry the data collection after 22 seconds. cmd_type=extractor name=tap-zendesk run_id=97154896-0092-4a8e-8390-63d337f5bac6 state_id=2022-12-20T214210--tap-zendesk--target-jsonl stdio=stderr
2022-12-20T21:43:01.117622Z [info     ] time=2022-12-20 18:43:01 name=singer level=INFO message=METRIC: {"type": "timer", "metric": "http_request_duration", "value": 0.37802696228027344, "tags": {"endpoint": "https://woothemes.zendesk.com/api/v2/tickets/5796575/metrics.json", "status": "succeeded"}} cmd_type=extractor name=tap-zendesk run_id=97154896-0092-4a8e-8390-63d337f5bac6 state_id=2022-12-20T214210--tap-zendesk--target-jsonl stdio=stderr
``` 
# Risks
 - 
 
# Rollback steps
 - revert this branch
